core/trino-main/src/main/java/io/trino/operator/DynamicFilterSourceOperator.java

```diff
@@ -487,4 +319,178 @@ public boolean isFinished()
             return current == null && finished;
         }
     }
+
+    private class ChannelFilter
+    {
+        private final Type type;
+
+        private ChannelState state;
+        private boolean collectMinMax;
+        // May be dropped if the predicate becomes too large.
+        @Nullable
+        private BlockBuilder blockBuilder;
+        @Nullable
+        private TypedSet valueSet;
+        @Nullable
+        private Block minValues;
+        @Nullable
+        private Block maxValues;
+        @Nullable
+        private BlockPositionComparison minMaxComparison;
+
+        public ChannelFilter(Channel channel, boolean minMaxEnabled, PlanNodeId planNodeId, BlockTypeOperators blockTypeOperators, int channelIndex)
+        {
+            type = channels.get(channelIndex).type;
```

Comment:
Can we do `type = channel.type` and remove the `channelIndex` argument ?

Reply:
done

Reply:
There is still unused argument `channelIndex`